server.js

```diff
@@ -5728,6 +5728,62 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Opera add-ons integration
+camp.route(/^\/opera-add-ons\/(v|d|rating|stars|rating-count)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  const type = match[1];
+  const name = match[2];
+  const format = match[3];
+  const badgeData = getBadgeData('opera add-ons', data);
+  const url = 'https://addons.opera.com/en/extensions/details/' + name;
+  request(url, function(err, res, buffer) {
+    if (!err) {
+      try {
+        const cheerio = require('cheerio');
+        const $ = cheerio.load(buffer);
+        const version = $('section.about > dl > dd').eq(2).text();
```

Comment:
This seems very unmaintainable and will break if Opera ever change their HTML even slightly. Do Opera not have an API for retrieving this data?